extensions/notification/NotificationConfigManager.js

```diff
@@ -1,283 +1,87 @@
-     * @returns {undefined}
+     * Add/update bucket notification configuration.
+     * Only supported in the Zookeeper backend.
+     *
+     * @param {String} bucket - bucket
+     * @param {Object} config - bucket notification configuration
+     * @return {boolean} - true if set
      */
-    _setMetastoreChangeStream() {
-        /**
-         * To avoid processing irrelevant events
-         * we filter by the operation types and
-         * only project the fields needed
-         */
-        const changeStreamPipeline = [
-            {
-                $match: {
-                    $or: [
-                        { operationType: 'delete' },
-                        { operationType: 'replace' },
-                        { operationType: 'update' },
-                    ]
-                }
-            },
-            {
-                $project: {
-                    '_id': 1,
-                    'operationType': 1,
-                    'documentKey._id': 1,
-                    'fullDocument._id': 1,
-                    'fullDocument.value.notificationConfiguration': 1
-                },
-            },
-        ];
-        this._metastoreChangeStream = new ChangeStream({
-            logger: this._logger,
-            collection: this._metastore,
-            pipeline: changeStreamPipeline,
-            handler: this._handleChangeStreamChangeEvent.bind(this),
-            throwOnError: false,
-            useStartAfter: semver.gte(this._mongoVersion, '4.2.0'),
-        });
-        // start watching metastore
-        this._metastoreChangeStream.start();
+    setConfig(bucket, config) {
+        if (this._usesZookeeperBackend) {
+            return this._configManagerBackend.setConfig(bucket, config);
+        }
+        return false;
     }
 
     /**
-     * Sets up the NotificationConfigManager by
-     * connecting to mongo and initializing the
-     * change stream
-     * @param {Function} cb callback
-     * @returns {undefined}
+     * Remove bucket notification configuration
+     * Only supported in the Zookeeper backend.
+     *
+     * @param {String} bucket - bucket
+     * @return {undefined}
      */
-    setup(cb) {
-        this._setupMongoClient(err => {
-            if (err) {
-                this._logger.error('An error occured while setting up mongo client', {
-                    method: 'NotificationConfigManager.setup',
-                });
-                return cb(err);
-            }
-            try {
-                this._setMetastoreChangeStream();
-            } catch (error) {
-                this._logger.error('An error occured while establishing the change stream', {
-                    method: 'NotificationConfigManager._setMetastoreChangeStream',
-                });
-                return cb(error);
-            }
-            return cb();
-        });
+    removeConfig(bucket) {
+        if (this._usesZookeeperBackend) {
```

Comment:
not needed : implement the method in mongo backend as well, and just return false